### PR TITLE
[Issue-2615] Optimized travis build for snapshot artifacts publishing

### PR DESCRIPTION
**Pull request by vijikarthi**
**Change log description**
This PR address the issue https://github.com/pravega/pravega/issues/2615

**Purpose of the change**
To improve travis build time for the master brach when snapshots are require to be published 

**What the code does**
The snapshot build stage perform publishing artifacts only after the `master` branch is successfully compiled and tested. 

**How to verify it**
Tested and confirmed on Travis for the PR branch
https://travis-ci.org/pravega/pravega/builds/382492364

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,23 +20,21 @@ jobs:
...
-
     - stage: build
       if: type IN (push, pull_request)
       install: ./gradlew assemble
       script: ./gradlew check
       after_success:
            - bash <(curl -s https://codecov.io/bash) -t ccceafaf-7c60-4a02-9165-480174b535a2
 
+    - stage: snapshot
+      if: branch = master
```
Comment (Prabha-Veerubhotla) [inline .travis.yml]: Can you please make it `if: branch = master AND type != pull_request`. 
We don't want every commit to pull request, publish snapshots to jcenter.

Comment (Prabha-Veerubhotla) [reply .travis.yml]: No, please check the commits to PRs,  after your PR went into master, every commit has 2 build stages. for ex : this build https://travis-ci.org/pravega/pravega/builds/382607193?utm_source=github_status&utm_medium=notification
from PR https://github.com/pravega/pravega/pull/2618.

This is not yet merged into master, but it tries to publish snapshots to jcenter.

Comment (Prabha-Veerubhotla) [reply .travis.yml]: *  Every PR from fork branch to master of pravega, has event type `pull_request` , (continuous-integration/travis-ci/pr)  and the branch will be taken as master.
*  By default the event type of `pull_request` or `push` is considered  in the build stages, unless some condition is specified in `travis.yml`.

Comment (vijikarthi) [reply .travis.yml]: I think you you are right. The build stage is applying for PRs merging to master as well. Thanks for finding it and I will fix the issue.

**Pull request merged**